test/e2e/pod_initcontainers_test.go

```diff
@@ -167,4 +168,26 @@ var _ = Describe("Podman init containers", func() {
 		Expect(firstResult).ToNot(Equal(secondCheckLog.OutputToString()))
 	})
 
+	It("podman set restart policy of init containers to \"no\"", func() {
+		initContainerName := "init-container"
+		session := podmanTest.Podman([]string{"create", "--pod", "new:foobar", fedoraMinimal})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		// add a init container
+		session = podmanTest.Podman([]string{"create", "--pod", "foobar", "--name", initContainerName, "--init-ctr", "once", fedoraMinimal, "false"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		// start the pod
+		startPod := podmanTest.Podman([]string{"pod", "start", "foobar"})
+		startPod.WaitWithDefaultTimeout()
+		// Non-zero exit due to a failing init container
+		Expect(startPod).Should(Exit(125))
+
+		// check restart policy of the init container
+		checkRestartPolicy := podmanTest.Podman([]string{"inspect", initContainerName, "--format", "{{ .HostConfig.RestartPolicy.Name }}"})
+		checkRestartPolicy.WaitWithDefaultTimeout()
+		Expect(checkRestartPolicy).Should(gbytes.Say(define.RestartPolicyNo))
+	})
```

Comment:
This test failed in my local environment. It might means init containers added by `podman  create --init-ctr` doesn't have a restart policy name.

Reply:
Yeah, looks like restart policy is not added to an init container even when the `--restart` flag is set.

Reply:
Removed this test. So that, this PR only fix and test restart policy of init container created by `podman kube play`. Is it okay to handle this in another PR or issue?